client/packages/invoices/src/Prescriptions/LineEditView/LineEditView.tsx

```diff
@@ -51,9 +74,78 @@ export const PrescriptionLineEditView = () => {
     });
   }, [currentItem]);
 
-  if (isLoading) return <BasicSpinner />;
+  useConfirmOnLeaving(
+    isDirty,
+    // Need a custom checking method here, as we don't want to warn user when
+    // switching to a different item within this page
+    (current, next) => {
+      if (!isDirty) return false;
+
+      const currentPathParts = current.pathname.split('/');
+      const nextPathParts = next.pathname.split('/');
+      // Compare URLS, but don't include the last part, which is the ItemID
+      currentPathParts.pop();
+      nextPathParts.pop();
+      return !isEqual(currentPathParts, nextPathParts);
+    }
+  );
+
+  const updateAllLines = (lines: DraftStockOutLine[]) => {
+    if (itemId === 'new') {
+      newItemId.current = lines[0]?.item.id;
+    }
+
+    if (typeof itemId === 'string')
+      setAllDraftLines({ ...allDraftLines, [itemId]: lines });
+  };
+
+  const onSave = async () => {
+    if (!isDirty) return;
+
+    const flattenedLines = Object.values(allDraftLines).flat();
+
+    // needed since placeholders aren't being created for prescriptions yet, but
+    // still adding to array
+    const isOnHold = flattenedLines.some(
+      ({ stockLine, location }) => stockLine?.onHold || location?.onHold
```

Comment:
onHold will just not be queried in future I think, when we do the new stockline selector it won't be shown...